tfhe/src/high_level_api/strings/ascii/no_pattern.rs

```diff
@@ -0,0 +1,331 @@
+use crate::high_level_api::global_state::with_internal_keys;
+use crate::high_level_api::integers::FheUint16;
+use crate::high_level_api::keys::InternalServerKey;
+use crate::high_level_api::strings::ascii::FheAsciiString;
+use crate::prelude::FheStringRepeat;
+use crate::strings::ciphertext::UIntArg;
+use crate::strings::client_key::EncU16;
+use crate::{FheBool, Tag};
+
+pub enum FheStringLen {
+    NoPadding(u16),
+    Padding(FheUint16),
+}
```

Comment:
Couldn't we use a trivial ciphertext here in the NoPadding case instead of having an enum ? Maybe it would make it easier to write generic code without having to match everywhere ?

Reply:
Its the design used in integers

I depends on how explicit we want the thing to be, if we always return a FheUint16 then we lose the fact that the len can be trivially known appears with an explicit type to the user.

But maybe it would indeed be better as I think most of the time the lengh is going to be non-trivial et even if you start with a string wihout padding, some ops will cause them to have padding

Reply:
Yes indeed, it depends on how people plan to use it